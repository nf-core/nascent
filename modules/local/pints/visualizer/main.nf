process PINTS_VISUALIZER {
    tag "${meta.id}"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    // NOTE Stopped publishing at 1.1.9 https://quay.io/repository/biocontainers/pypints?tab=tags
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/f1/f1a9e30012e1b41baf9acd1ff94e01161138d8aa17f4e97aa32f2dc4effafcd1/data'
        : 'community.wave.seqera.io/library/pybedtools_bedtools_htslib_pip_pypints:39699b96998ec5f6'}"

    input:
    tuple val(meta), path(bam)
    val assay_type

    output:
    tuple val(meta), path("*_pl.bw"), emit: plus_bw
    tuple val(meta), path("*_mn.bw"), emit: minus_bw
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    // def chr_arg = chr_name ? "--chrom ${chr_name}" : ''
    // def norm_arg = norm_factor != null ? "--norm-fact ${norm_factor}" : ''
    // def rpm_arg = rpm_normalize ? "--rpm" : ''
    // def rc_arg = reverse_complement ? "--reverse-complement" : ''
    """
    pints_visualizer \\
        --bam ${bam} \\
        --exp-type ${assay_type} \\
        --output-prefix ${prefix} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
        pints: \$(pints_visualizer --version)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_${chr_name}_plus.bigwig
    touch ${prefix}_${chr_name}_minus.bigwig

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
        pints: \$(pints_visualizer --version)
    END_VERSIONS
    """
}
