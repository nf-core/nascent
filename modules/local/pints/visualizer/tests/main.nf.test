nextflow_process {

    name "Test Process PINTS_VISUALIZER"
    script "../main.nf"
    process "PINTS_VISUALIZER"
    tag "modules"
    tag "modules_local"
    tag "pints"
    tag "pints/visualizer"

    test("Should run PINTS_VISUALIZER with PROseq data and produce strand-specific bigwigs") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample', single_end:false ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    'chr1'
                ])
                input[1] = 'PROseq'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.plus_bw.size() == 1 },
                { assert process.out.minus_bw.size() == 1 },
                { assert snapshot(
                    process.out.plus_bw.get(0)[1],
                    process.out.minus_bw.get(0)[1]
                    ).match()
                },
                { assert snapshot(path(process.out.versions[0]).yaml).match("flipped_versions") }
            )
        }
    }

    test("Should run PINTS_VISUALIZER with GROseq data without reverse complement") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_gro', single_end:false ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    'chr1'
                ])
                input[1] = 'GROseq'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.plus_bw.size() == 1 },
                { assert process.out.minus_bw.size() == 1 },
                { assert snapshot(
                    process.out.plus_bw.get(0)[1],
                    process.out.minus_bw.get(0)[1]
                    ).match()
                },
                { assert snapshot(path(process.out.versions[0]).yaml).match("versions") }
            )
        }
    }
}
