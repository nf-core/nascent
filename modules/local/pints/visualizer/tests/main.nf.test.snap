{
    "versions": {
        "content": [
            {
                "PINTS_VISUALIZER": {
                    "python": "3.12.6",
                    "pints": "1.1.13"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-08T18:07:16.978238"
    },
    "flipped_versions": {
        "content": [
            {
                "PINTS_VISUALIZER": {
                    "python": "3.12.6",
                    "pints": "1.1.13"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-08T18:07:07.331434"
    },
    "Should run PINTS_VISUALIZER with PROseq data and produce strand-specific bigwigs": {
        "content": [
            "test_sample_pl.bw:md5,6fd3346be91cc84750f839b050c6bc5b",
            "test_sample_mn.bw:md5,9eecf5936bdf1c0c44ec123636a305f5"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-04T18:24:13.206646"
    },
    "Should run PINTS_VISUALIZER with GROseq data without reverse complement": {
        "content": [
            "test_gro_pl.bw:md5,e399e77f224d993e789bf8f7c4b826e1",
            "test_gro_mn.bw:md5,21c7ef549d8cb39788e13080dc573848"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-05-04T18:24:22.747334"
    }
}